src/main/java/org/sonatype/repository/vgo/internal/hosted/VgoHostedFacetImpl.java

```diff
@@ -0,0 +1,239 @@
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.StreamPayload;
+import org.sonatype.nexus.repository.view.payloads.StreamPayload.InputStreamSupplier;
+import org.sonatype.nexus.transaction.Transactional;
+import org.sonatype.nexus.transaction.UnitOfWork;
+import org.sonatype.repository.vgo.VgoAssetKind;
+import org.sonatype.repository.vgo.internal.metadata.VgoAttributes;
+import org.sonatype.repository.vgo.internal.metadata.VgoInfo;
+import org.sonatype.repository.vgo.internal.util.CompressedContentExtractor;
+import org.sonatype.repository.vgo.internal.util.VgoDataAccess;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.impl.io.EmptyInputStream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.view.Payload.UNKNOWN_SIZE;
+import static org.sonatype.repository.vgo.VgoAssetKind.VGO_MODULE;
+import static org.sonatype.repository.vgo.internal.util.VgoDataAccess.HASH_ALGORITHMS;
+
+/**
+ * Vgo hosted implementation
+ *
+ * @since 0.0.2
+ */
+@Named
+public class VgoHostedFacetImpl
+    extends FacetSupport
+  implements VgoHostedFacet
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final VgoDataAccess vgoDataAccess;
+
+  @Override
+  protected void doInit(final Configuration configuration) throws Exception {
+    super.doInit(configuration);
+  }
+
+  @Inject
+  public VgoHostedFacetImpl(final VgoDataAccess vgoDataAccess) {
+    this.vgoDataAccess = checkNotNull(vgoDataAccess);
+  }
+
+  @Override
+  @Nullable
+  @Transactional
+  public Content getInfo(final String path,
+                         final VgoAttributes vgoAttributes) {
+    checkNotNull(path);
+    String newPath = path.replaceAll("\\.info", "\\.zip");
+
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset asset = vgoDataAccess.findAsset(tx, tx.findBucket(getRepository()), newPath);
+    if (asset == null) {
+      return null;
+    }
+
+    StreamPayload streamPayload = new StreamPayload(
+        () -> doGetInfo(asset, vgoAttributes),
+        UNKNOWN_SIZE,
+        ContentTypes.APPLICATION_JSON);
+    return new Content(streamPayload);
+  }
+
+  @Transactional
+  @Override
+  public Content getList(final String module) {
```

Comment:
I like the approach you've used for all of the metadata here. Hopefully this reduces any HA issues and makes it easier to scale the go format 👍